### PR TITLE
Added OpenStreetMap layer to map (#215)

### PR DESCRIPTION
**Pull request by mitjap**
#215

### CODE REVIEW

File: piplmesh/panels/map/static/piplmesh/panel/map/panel.js [hunk 1]
```diff
@@ -1,5 +1,7 @@
 $(document).ready(function () {
+    var element = document.getElementById('map_canvas');
```
Comment (mitar) [inline piplmesh/panels/map/static/piplmesh/panel/map/panel.js]: Why do you need this variable? You are using it only once? And you can use jQuery syntax.

File: piplmesh/panels/map/static/piplmesh/panel/map/panel.js [hunk 2]
```diff
@@ -7,17 +9,37 @@ $(document).ready(function () {
         navigationControl: false,
         scaleControl: false,
         draggable: false,
-        mapTypeId: google.maps.MapTypeId.ROADMAP,
+        mapTypeId: "OSM",
```
Comment (mitar) [inline piplmesh/panels/map/static/piplmesh/panel/map/panel.js]: Constant strings should be in `'`.

Comment (mitar) [inline piplmesh/panels/map/static/piplmesh/panel/map/panel.js]: Do you really need to use acronym? `OpenStreetMap` would be much easier to understand.

File: piplmesh/panels/map/static/piplmesh/panel/map/panel.js [hunk 3]
```diff
@@ -7,17 +9,37 @@ $(document).ready(function () {
...
+        mapTypeControlOptions: {
+            mapTypeIds: [
```
Comment (mitar) [inline piplmesh/panels/map/static/piplmesh/panel/map/panel.js]: Why you are at it. Put all object/dict keys into strings. So `mapTypeIds:` -> `'mapTypeIds':`

File: piplmesh/panels/map/static/piplmesh/panel/map/panel.js [hunk 4]
```diff
@@ -7,17 +9,37 @@ $(document).ready(function () {
...
         position: nodeLocation,
         map: map,
         title: node.name
     });
+
+    var nodeName = $('<p/>').text(node.name).append(' | ');
+    var nodeWebsite = $('<a/>').prop('href', node.url).text(gettext("more info"));
+    nodeName.append(nodeWebsite);
+    $('#map_info').append(nodeName);
+
```
Comment (mitar) [inline piplmesh/panels/map/static/piplmesh/panel/map/panel.js]: Unnecessary line.

File: piplmesh/panels/map/static/piplmesh/panel/map/panel.js [hunk 5]
```diff
@@ -7,17 +9,37 @@ $(document).ready(function () {
...
-    var map = new google.maps.Map(document.getElementById('map_canvas'), myOptions);
-    var nodeName = $('<p/>').text(node.name).append(' | ');
-    var nodeWebsite = $('<a/>').prop('href', node.url).text(gettext("more info"));
-    nodeName.append(nodeWebsite);
-    $('#map_info').append(nodeName);
+
+    var map = new google.maps.Map(element, myOptions);
+
+    map.mapTypes.set("OSM", new google.maps.ImageMapType({
+        getTileUrl: function(coord, zoom) {
```
Comment (mitar) [inline piplmesh/panels/map/static/piplmesh/panel/map/panel.js]: Write out variable names (`coord`).

File: piplmesh/panels/map/static/piplmesh/panel/map/panel.js [hunk 6]
```diff
@@ -7,17 +9,37 @@ $(document).ready(function () {
...
+            return "http://tile.openstreetmap.org/" + zoom + "/" + coord.x + "/" + coord.y + ".png";
+        },
+        tileSize: new google.maps.Size(256, 256),
+        name: "OpenStreetMap",
```
Comment (mitar) [inline piplmesh/panels/map/static/piplmesh/panel/map/panel.js]: Constant string.

File: piplmesh/panels/map/static/piplmesh/panel/map/panel.js [hunk 7]
```diff
@@ -7,17 +9,37 @@ $(document).ready(function () {
...
-    var nodeName = $('<p/>').text(node.name).append(' | ');
-    var nodeWebsite = $('<a/>').prop('href', node.url).text(gettext("more info"));
-    nodeName.append(nodeWebsite);
-    $('#map_info').append(nodeName);
+
+    var map = new google.maps.Map(element, myOptions);
+
+    map.mapTypes.set("OSM", new google.maps.ImageMapType({
+        getTileUrl: function(coord, zoom) {
+            return "http://tile.openstreetmap.org/" + zoom + "/" + coord.x + "/" + coord.y + ".png";
```
Comment (mitar) [inline piplmesh/panels/map/static/piplmesh/panel/map/panel.js]: Constant strings.

File: piplmesh/panels/map/static/piplmesh/panel/map/panel.js [hunk 8]
```diff
@@ -1,23 +1,43 @@
...
+
+    var marker = new google.maps.Marker({
+        'position': nodeLocation,
+        'map': map,
+        'title': node.name
+    });
+
     var nodeName = $('<p/>').text(node.name).append(' | ');
-    var nodeWebsite = $('<a/>').prop('href', node.url).text(gettext("more info"));
+    var nodeWebsite = $('<a/>').prop('href', node.url).text(gettext('more info'));
```
Comment (mitar) [inline piplmesh/panels/map/static/piplmesh/panel/map/panel.js]: No, user exposed strings we have in `"`. That's the main difference. The idea is that you can fast search for all strings which are visible to users. And other strings which are used internally.

File: piplmesh/panels/map/static/piplmesh/panel/map/panel.js [hunk 9]
```diff
@@ -1,23 +1,43 @@
...
-        draggable: false,
-        mapTypeId: google.maps.MapTypeId.ROADMAP,
-        streetViewControl: false
+        'zoom': 15,
+        'center': nodeLocation,
+        'scrollwheel': false,
+        'navigationControl': false,
+        'scaleControl': false,
+        'draggable': false,
+        'mapTypeId': 'OpenStreetMap',
```
Comment (mitar) [inline piplmesh/panels/map/static/piplmesh/panel/map/panel.js]: Can you leave roadmap as default map?

Comment (mitar) [inline piplmesh/panels/map/static/piplmesh/panel/map/panel.js]: You can open a ticket/add TODO that user's could configure the default map.

**Comment by mitjap** [comment]
Added TODO, switched default layer and fixed quotes

**Comment by mitar** [comment]
I cleaned code a bit. Check what I changed.

**Comment by mitjap** [comment]
Changes reviewed :)

**Pull request merged**